Apps/Src/UpdateVelocity.c

```diff
@@ -47,7 +52,12 @@ void Task_UpdateVelocity(void *p_arg)
         if(brakePedalPercent >= UNTOUCH_PEDALS_PERCENT){
             desiredVelocity = 0;
             desiredMotorCurrent = 0;
-        } else {
+        }
+        else if(Switches_Read(REGEN_SW) && (MotorController_ReadVelocity() > THRESHOLD_VEL && RegenEnable)){ 
+            desiredVelocity = 0;
+            desiredMotorCurrent = REGEN_CURRENT;
+        } 
+        else {
             if(!Switches_Read(REV_SW)){
                 desiredVelocity = MAX_VELOCITY;
```

Comment:
I would move this `MAX_VELOCITY` macro within this file rather than leaving it in whatever header it is in currently. It shouldn't be used anywhere else.